opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/servlet/SpanDecorator.java

```diff
@@ -0,0 +1,59 @@
+package io.opentracing.contrib.spring.web.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+
+/**
+ * SpanDecorator to decorate span at different stages in filter processing.
+ *
+ * @author Pavol Loffay
+ */
+public interface SpanDecorator {
+
+    /**
+     * This is called in
+     * {@link org.springframework.web.servlet.HandlerInterceptor#preHandle(HttpServletRequest, HttpServletResponse, Object)}.
+     *
+     * @param httpServletRequest request
+     * @param handler handler
+     * @param span current span
+     */
+    void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span);
+
+    /**
+     * This is called in
+     * {@link org.springframework.web.servlet.HandlerInterceptor#afterCompletion(HttpServletRequest, HttpServletResponse, Object, Exception)}
+     *
+     * @param httpServletRequest request
+     * @param httpServletResponse response
+     * @param handler handler
+     * @param ex exception
+     * @param span current span
+     */
+    void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Object handler,
+                           Exception ex, Span span);
+
+    /**
+     * Adds standard tags to span. {@link Tags#HTTP_URL}, {@link Tags#HTTP_STATUS}, {@link Tags#HTTP_METHOD} and
+     * {@link Tags#COMPONENT}.
+     */
+    SpanDecorator STANDARD_TAGS = new SpanDecorator() {
+        @Override
+        public void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span) {
+            Tags.COMPONENT.set(span, "java-spring-web-interceptor");
+
+            Tags.HTTP_METHOD.set(span, httpServletRequest.getMethod());
+            //without query params
+            Tags.HTTP_URL.set(span, httpServletRequest.getRequestURL().toString());
+        }
+
+        @Override
+        public void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                                      Object handler, Exception ex, Span span) {
+            Tags.HTTP_STATUS.set(span, httpServletResponse.getStatus());
```

Comment:
I meant not to base the decision about adding an error ta on the status code but only on the presence of the exception. This way you avoid having to decide if 404 is an error or just a nice REST response for "record not found"